src/lib/ast/skillExpresssion.spec.ts

```diff
@@ -0,0 +1,280 @@
+import { Skill } from "../types";
+import { Or } from "./or";
+import { And } from "./and";
+import { SkillExpression } from "./skillExpression";
+import { parseJsonExpression } from "./jsonHandler";
+import { Variable } from "./variable";
+import { SkillsRelations } from "./skillsRelation";
+import { Empty } from "./empty";
+
+describe("precondition formula", () => {
+
+	const skill1 = { id: "skill:1", repositoryId: "Map 1", nestedSkills: [] };
+	const skill2 = { id: "skill:2", repositoryId: "Map 1", nestedSkills: [] };
+	const skill3 = { id: "skill:3", repositoryId: "Map 1", nestedSkills: [] };
+	const skill4 = { id: "skill:4", repositoryId: "Map 1", nestedSkills: [] };
+	const skill5 = { id: "skill:5", repositoryId: "Map 1", nestedSkills: [] };
+	const skill6 = { id: "skill:6", repositoryId: "Map 1", nestedSkills: [] };
+	const skill7 = { id: "skill:7", repositoryId: "Map 1", nestedSkills: [] };
+	const skill8 = { id: "skill:8", repositoryId: "Map 1", nestedSkills: [] };
+	const skill9 = { id: "skill:9", repositoryId: "Map 1", nestedSkills: ["skill:7", "skill:8"] };
+	const skill10 = { id: "skill:10", repositoryId: "Map 1", nestedSkills: 
+    ["skill:1", "skill:2", "skill:3", "skill:4", "skill:5", "skill:6"] };
+
+	const skills: Skill[] = [skill1, skill2, skill3, skill4,
+		skill5, skill6, skill7, skill8, skill9, skill10];
+
+	const skillsRelations = new SkillsRelations(skills);
+
+	it("skillExpression type", () => {
+
+        const emptySkills = new Empty([]);
+        const orSkills = new Or([new Variable(skill1)]);
+        const andSkills = new And([new Variable(skill2)]);
+		const variable = new Variable(skill3);
+
+        expect(emptySkills.getExpressionType()).toBe("Empty");
+        expect(orSkills.getExpressionType()).toBe("Or");
+        expect(andSkills.getExpressionType()).toBe("And");
+		expect(variable.getExpressionType()).toBe("Variable");
+
+	});
+
+	it("empty expression, no required skill for the expression", () => {
+
+        const emptySkills = new Empty([]);
+
+        expect(emptySkills.evaluate([], skillsRelations)).toBeTruthy();
+        expect(emptySkills.extractSkills().length).toBe(0);
+		expect(emptySkills.toJson()).toBe("");
+	});
+
+	it("skills in precondition formula", () => {

```

Comment:
Usually, tests should be as atomic as possible -> Test only one case per function, aggregate similar tests to one suite (describe), use meaning ful naming.

I would recommend a structure like this (may also organized in other hierarchies):
```typescript
describe("precondition formula", () => {
    ...
    describe("AND", () => {
    ...
        it("Top-Level skills; No nested expressions; fulfilled -> evaluates true)", () => {
            ...
        }
        it("Top-Level skills; No nested expressions; not fulfilled -> evaluates false)", () => {
            ...
        }
        it("Nested skills; No nested expressions; fulfilled -> evaluates true)", () => {
            ...
        }
        ...
        it("Top-Level skills; Nested expressions; fulfilled -> evluates true)", () => {
            ...
        }
        ...
    }
    describe("OR", () => {
        ...
    }
}
```